src/rpc/blockchain.cpp

```diff
@@ -2836,6 +2836,15 @@ static RPCHelpMan loadtxoutset()
             strprintf("The base block header (%s) must appear in the headers chain. Make sure all headers are syncing, and call this RPC again.",
                       base_blockhash.ToString()));
     }
+
+    bool start_block_invalid = WITH_LOCK(::cs_main,
+            return snapshot_start_block->nStatus & BLOCK_FAILED_MASK);
+    if (start_block_invalid) {
+        throw JSONRPCError(
+            RPC_INTERNAL_ERROR,
+            "The base block header is part of an invalid chain.");
```

Comment:
Nit: Would it be better to include the base block hash on the error message to help in debugging 

```cpp
throw JSONRPCError(
            RPC_INTERNAL_ERROR,
            "The base block header (%s) is part of an invalid chain.", base_blockhash.ToString());
```

Reply:
taken with minor edits